p2p/nat/nat.go

```diff
@@ -243,7 +247,13 @@ func (nat *NAT) addMapping(m *mapping) {
 	nat.proc.AddChild(m.proc)
 
 	nat.mappingmu.Lock()
-	nat.mappings = append(nat.mappings, m)
+	nat.mappings[m] = struct{}{}
+	nat.mappingmu.Unlock()
+}
+
+func (nat *NAT) rmMapping(m *mapping) {
```

Comment:
not anymore, the natted addresses arent added to our address book, they're calculated at the last possible moment (making them not stale): 
- https://github.com/jbenet/go-ipfs/blob/master/p2p/host/basic/basic_host.go#L217-L231
- we should be using `host.Addrs()` to give people most up-to-date view into our addresses.
  we should move away from keeping our own addrs in the address book as things go stale :(